code/modules/integrated_electronics/subtypes/lists.dm

```diff
@@ -138,9 +138,11 @@
 	var/list/input_list = get_pin_data(IC_INPUT, 1)
 	var/index = get_pin_data(IC_INPUT, 2)
 	var/item = get_pin_data(IC_INPUT, 3)
-	if(!islist(item))				//crh proof
-		input_list[index] = item
-		set_pin_data(IC_OUTPUT, 1, input_list)
+	var/list/red_list = list()
```

Comment:
you don't need to create an empty list here if you're going to copy the other list